storage/src/mmr/proof.rs

```diff
@@ -165,11 +174,20 @@ impl<D: Digest> Proof<D> {
         if elements.is_empty() {
             return self.size == 0 && *root == hasher.root(0, core::iter::empty());
         }
+        if !PeakIterator::check_validity(self.size) {
+            return false;
+        }
 
         // Single pass to collect all required positions with deduplication
         let mut node_positions = BTreeSet::new();
         let mut nodes_required = BTreeMap::new();
+
         for (_, loc) in elements {
+            if leaf_loc_to_pos(*loc) >= self.size {
```

Comment:
Nice catch 👍

Reply:
yup, nodes_required_for_range_proof will panic on out-of-bounds input loc.